test/test_dbc.cpp

```diff
@@ -125,12 +155,66 @@ TEST_CASE("Special characters in unit") {
 
 
     auto parser = libdbc::DbcParser();
-    parser.parse_file(std::string(filename));
+    parser.parse_file(filename);
 
     REQUIRE(parser.get_messages().size() == 1);
-    REQUIRE(parser.get_messages().at(0).signals.size() == 1);
+    REQUIRE(parser.get_messages().at(0).signals().size() == 1);
     SECTION("Checking that signal with special characters as unit is parsed correctly") {
-      const auto signal = parser.get_messages().at(0).signals.at(0);
+      const auto signal = parser.get_messages().at(0).signals().at(0);
       REQUIRE(signal.unit.compare("Km/h") == 0);
     }
 }
+
+TEST_CASE("Parse Message little endian") {
+    const auto* filename = std::tmpnam(NULL);
+
+    auto* file = std::fopen(filename, "w");
+    CHECK(file);
+
+    std::fputs(PRIMITIVE_DBC.c_str(), file);
+    std::fputs(R"(BO_ 541 STATUS: 8 DEVICE1
+ SG_ Temperature : 48|16@1+ (0.01,-40) [-40|125] "C"  DEVICE1
+ SG_ SOH : 0|16@1+ (0.01,0) [0|100] "%"  DEVICE1
+ SG_ SOE : 32|16@1+ (0.01,0) [0|100] "%"  DEVICE1
+ SG_ SOC : 16|16@1+ (0.01,0) [0|100] "%"  DEVICE1)", file);
+    std::fclose(file);

```

Comment:
```
    auto* file = std::fopen(filename, "w");
    CHECK(file);

    std::fputs(PRIMITIVE_DBC.c_str(), file);
    std::fputs(R"(BO_ 541 STATUS: 8 DEVICE1
 SG_ Temperature : 48|16@1+ (0.01,-40) [-40|125] "C"  DEVICE1
 SG_ SOH : 0|16@1+ (0.01,0) [0|100] "%"  DEVICE1
 SG_ SOE : 32|16@1+ (0.01,0) [0|100] "%"  DEVICE1
 SG_ SOC : 16|16@1+ (0.01,0) [0|100] "%"  DEVICE1)", file);
    std::fclose(file);
```

This might be able to use the : 
`create_tmp_dbc_with(filename, R"(B...));`

Same with some of the other tests